External/FEXCore/Source/Interface/Core/JIT/x86_64/JIT.cpp

```diff
@@ -628,137 +625,160 @@ void *JITCore::CompileCode([[maybe_unused]] FEXCore::IR::IRListView<true> const
 	void *Entry = getCurr<void*>();
   this->IR = IR;
 
-  LogMan::Throw::A(RAPass->HasFullRA(), "Needs RA");
+  if (CTX->GetGdbServerStatus()) {
+    Label RunBlock;
 
-  SpillSlots = RAPass->SpillSlots();
+    // If we have a gdb server running then run in a less efficient mode that checks if we need to exit
+    // This happens when single stepping
+    static_assert(sizeof(CTX->Config.RunningMode) == 4, "This is expected to be size of 4");
+    mov(rax, qword [STATE + (offsetof(FEXCore::Core::InternalThreadState, CTX))]);
 
-  if (SpillSlots) {
-    sub(rsp, SpillSlots * 16 + 8);
-  }
-  else {
-    sub(rsp, 8);
-  }
+    // If the value == 0 then branch to the top
+    cmp(dword [rax + (offsetof(FEXCore::Context::Context, Config.RunningMode))], 0);
+    je(RunBlock);
+    // Else we need to pause now
+    mov(rax, ThreadPauseHandlerAddress);
+    jmp(rax);
+    ud2();
 
-#ifdef BLOCKSTATS
-  BlockSamplingData::BlockData *SamplingData = CTX->BlockData->GetBlockData(HeaderOp->Entry);
-  if (GetSamplingData) {
-    mov(rcx, reinterpret_cast<uintptr_t>(SamplingData));
-    rdtsc();
-    shl(rdx, 32);
-    or(rax, rdx);
-    mov(qword [rcx + offsetof(BlockSamplingData::BlockData, Start)], rax);
+    L(RunBlock);
   }
 
-  auto ExitBlock = [&]() {
+  if (HeaderOp->ShouldInterpret) {
```

Comment:
Same comment as the AArch64 side.

Reply:
Because of gdb stub header, and later on of unsynchronized (implicit) RIP